packages/build-utils/src/fs/run-user-scripts.ts

```diff
@@ -234,55 +242,86 @@ export async function getNodeVersion(
 
 export async function scanParentDirs(
   destPath: string,
-  readPackageJson = false
+  {
+    readPackageJson = false,
+    spawnOpts,
+  }: {
+    readPackageJson?: boolean;
+    spawnOpts?: SpawnOptions;
+  } = {}
 ): Promise<ScanParentDirsResult> {
   assert(path.isAbsolute(destPath));
 
   const pkgJsonPath = await walkParentDirs({
     base: '/',
     start: destPath,
     filename: 'package.json',
+    predicate: fullpath => {
+      const data = fs.readFileSync(fullpath, 'utf-8');
+      const json = JSON.parse(data);
+      return type json.packageManager === 'string';
```

Comment:
I think this is wrong depending on the caller because only the case of corepack would care about the `packageManager` prop.

The code as written would always return false for every package.json if they don't have `packageManager` prop.

Reply:
true